runtime/queries/nix/highlights.scm

```diff
@@ -1,81 +1,88 @@
  (#is-not? local))
 
 ((identifier) @function.builtin
  (#match? @function.builtin "^(__add|__addErrorContext|__all|__any|__appendContext|__attrNames|__attrValues|__bitAnd|__bitOr|__bitXor|__catAttrs|__compareVersions|__concatLists|__concatMap|__concatStringsSep|__deepSeq|__div|__elem|__elemAt|__fetchurl|__filter|__filterSource|__findFile|__foldl'|__fromJSON|__functionArgs|__genList|__genericClosure|__getAttr|__getContext|__getEnv|__hasAttr|__hasContext|__hashFile|__hashString|__head|__intersectAttrs|__isAttrs|__isBool|__isFloat|__isFunction|__isInt|__isList|__isPath|__isString|__langVersion|__length|__lessThan|__listToAttrs|__mapAttrs|__match|__mul|__parseDrvName|__partition|__path|__pathExists|__readDir|__readFile|__replaceStrings|__seq|__sort|__split|__splitVersion|__storePath|__stringLength|__sub|__substring|__tail|__toFile|__toJSON|__toPath|__toXML|__trace|__tryEval|__typeOf|__unsafeDiscardOutputDependency|__unsafeDiscardStringContext|__unsafeGetAttrPos|__valueSize|abort|baseNameOf|derivation|derivationStrict|dirOf|fetchGit|fetchMercurial|fetchTarball|fromTOML|import|isNull|map|placeholder|removeAttrs|scopedImport|throw|toString)$")
  (#is-not? local))
 
 [
-  (string)
-  (indented_string)
+  (string_expression)
+  (indented_string_expression)
 ] @string
 
 [
-  (path)
-  (hpath)
-  (spath)
+  (path_expression)
+  (hpath_expression)
+  (spath_expression)
 ] @string.special.path
 
-(uri) @string.special.uri
+(uri_expression) @string.special.uri
 
-; boolean
-((identifier) @constant.builtin.boolean (#match? @constant.builtin.boolean "^(true|false)$")) @constant.builtin.boolean
-; null
-((identifier) @constant.builtin (#eq? @constant.builtin "null")) @constant.builtin
-
-(integer) @constant.numeric.integer
-(float) @constant.numeric.float
+[
+  (integer_expression)
+  (float_expression)
+] @number
 
 (interpolation
   "${" @punctuation.special
   "}" @punctuation.special) @embedded
 
-(escape_sequence) @constant.character.escape
+(escape_sequence) @escape
 
-(function
+(function_expression
   universal: (identifier) @variable.parameter
 )
 
 (formal
   name: (identifier) @variable.parameter
   "?"? @punctuation.delimiter)
 
-(app
+(select_expression
+  attrpath: (attrpath (identifier)) @property)
+
+(apply_expression
   function: [
-    (identifier) @function
-    (select
+    (variable_expression (identifier)) @function
+    (select_expression
       attrpath: (attrpath
-        attr: (attr_identifier) @function .))])
-
+        attr: (identifier) @function .))])
 
-(unary
+(unary_expression
   operator: _ @operator)
 
-(binary
+(binary_expression
   operator: _ @operator)
 
-(attr_identifier) @variable.other.member
-(inherit attrs: (attrs_inherited (identifier) @variable.other.member) )
+(variable_expression (identifier) @variable)
+
+(binding
+  attrpath: (attrpath (identifier)) @property)
+
+(identifier) @property
+
+(inherit_from attrs: (inherited_attrs attr: (identifier) @property) )
```

Comment:
This pattern is redundant because the prior stanza (L77) captures all `identifier`s as `@property` (`@variable.other.member`) - precedence is top-to-bottom in helix. We should be able to just remove it.

Reply:
I just copied the file directly from upstream:
https://github.com/cstrahan/tree-sitter-nix/blob/master/queries/highlights.scm

Perhaps it would be best to fix there?

Reply:
Queries in tree-sitter grammar repositories tend to limit themselves to the scopes given by tree-sitter-cli ([these](https://github.com/tree-sitter/tree-sitter/blob/6192d2961096bda537e8ecb20de9c9dbd506cad8/cli/src/highlight.rs#L150-L171)), so there's usually some tailoring to do to use helix-specific scopes like how we separate floats and integers. The redundant stanzas can safely be removed upstream but not most of the other custom changes we carry.